File: codes/perl/conditionals.pl
=begin
Relational Operators / Comparison Operators:
equal to [==, eq]
not equal to [!=, ne]
greater than [>, gt]
less than [<, lt]
greater then or equal to [>=, ge]
less then or equal to [<=, le]

Logical Operators:
Logical AND [&&, and]
Logical OR [||, or]
Logical NOT [!, not]
=cut

sub print_bool {
    my ($val) = @_;
    return $val ? "true" : "false";
}

my $CORRECT_ANSWER = 100;

my $my_age;
my $my_answer;
my $has_driving_license;
my $can_drive;

print("\n# equal to [==, eq]\n");

$my_answer = "100";
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer == \$CORRECT_ANSWER): " . "\n");
if ($my_answer == $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = "100";
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer eq \$CORRECT_ANSWER): " . "\n");
if ($my_answer eq $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = 100;
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer == \$CORRECT_ANSWER): " . "\n");
if ($my_answer == $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = 100;
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer eq \$CORRECT_ANSWER): " . "\n");
if ($my_answer eq $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

print("\n# not equal to [!=, ne]\n");

$my_answer = "25";
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer != \$CORRECT_ANSWER): " . "\n");
if ($my_answer != $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = "25";
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer ne \$CORRECT_ANSWER): " . "\n");
if ($my_answer ne $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = 25;
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer != \$CORRECT_ANSWER): " . "\n");
if ($my_answer != $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

$my_answer = 25;
print("\$my_answer: " . $my_answer . "\n");
print("\$CORRECT_ANSWER: " . $CORRECT_ANSWER . "\n");
print("(\$my_answer ne \$CORRECT_ANSWER): " . "\n");
if ($my_answer ne $CORRECT_ANSWER) {
    print("Your answer is correct" . "\n");
} else {
    print("Your answer is wrong" . "\n");
}

print("\n# greater than [>, gt]\n");

$my_age = "70";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age > 50): " . "\n");
print("You are old" . "\n") if ($my_age > 50);

$my_age = "70";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age gt 50): " . "\n");
print("You are old" . "\n") if ($my_age gt 50);

$my_age = 70;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age > 50): " . "\n");
print("You are old" . "\n") if ($my_age > 50);

$my_age = 70;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age gt 50): " . "\n");
print("You are old" . "\n") if ($my_age gt 50);

print("\n# less than [<, lt]\n");

$my_age = "16";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age < 20): " . "\n");
print("You are young" . "\n") if ($my_age < 20);

$my_age = "16";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age lt 20): " . "\n");
print("You are young" . "\n") if ($my_age lt 20);

$my_age = 16;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age < 20): " . "\n");
print("You are young" . "\n") if ($my_age < 20);

$my_age = 16;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age lt 20): " . "\n");
print("You are young" . "\n") if ($my_age lt 20);

print("\n# greater then or equal to [>=, ge]\n");

$my_age = "73";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age >= 65): " . "\n");
print("You are allowed to retire" . "\n") if ($my_age >= 65);

$my_age = "73";
print("\$my_age: " . $my_age . "\n");
print("(\$my_age ge 65): " . "\n");
print("You are allowed to retire" . "\n") if ($my_age ge 65);

$my_age = 73;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age >= 65): " . "\n");
print("You are allowed to retire" . "\n") if ($my_age >= 65);

$my_age = 73;
print("\$my_age: " . $my_age . "\n");
print("(\$my_age ge 65): " . "\n");
print("You are allowed to retire" . "\n") if ($my_age ge 65);

print("\n# less then or equal to [<=, le]\n");

$my_age = "14"; 
print("\$my_age: " . $my_age . "\n");
print("(\$my_age <= 16): " . "\n");
print("You are not allowed to drive" . "\n") if ($my_age <= 16);

$my_age = "14"; 
print("\$my_age: " . $my_age . "\n");
print("(\$my_age le 16): " . "\n");
print("You are not allowed to drive" . "\n") if ($my_age le 16);

$my_age = 14; 
print("\$my_age: " . $my_age . "\n");
print("(\$my_age <= 16): " . "\n");
print("You are not allowed to drive" . "\n") if ($my_age <= 16);

$my_age = 14; 
print("\$my_age: " . $my_age . "\n");
print("(\$my_age le 16): " . "\n");
print("You are not allowed to drive" . "\n") if ($my_age le 16);

print("\n# Logical AND [&&, and]\n");

$my_age = 17;
print("\$my_age: " . $my_age . "\n");
$has_driving_license = 0;
print("\$has_driving_license: " . print_bool($has_driving_license) . "\n");
print("((\$my_age >= 17) && \$has_driving_license): " . "\n");
print(((($my_age >= 17) && $has_driving_license) ? "You are allowed to drive" . "\n" : "You are not allowed to drive" . "\n"));

$my_age = 17;
print("\$my_age: " . $my_age . "\n");
$has_driving_license = 1;
print("\$has_driving_license: " . print_bool($has_driving_license) . "\n");
print("((\$my_age >= 17) && \$has_driving_license): " . "\n");
print(((($my_age >= 17) && $has_driving_license) ? "You are allowed to drive" . "\n" : "You are not allowed to drive" . "\n"));

$my_age = 17;
print("\$my_age: " . $my_age . "\n");
$has_driving_license = 0;
print("\$has_driving_license: " . print_bool($has_driving_license) . "\n");
print("((\$my_age >= 17) and \$has_driving_license): " . "\n");
print(((($my_age >= 17) and $has_driving_license) ? "You are allowed to drive" . "\n" : "You are not allowed to drive" . "\n"));

$my_age = 17;
print("\$my_age: " . $my_age . "\n");
$has_driving_license = 1;
print("\$has_driving_license: " . print_bool($has_driving_license) . "\n");
print("((\$my_age >= 17) and \$has_driving_license): " . "\n");
print(((($my_age >= 17) and $has_driving_license) ? "You are allowed to drive" . "\n" : "You are not allowed to drive" . "\n"));

print("\n# Logical OR [||, or]\n");

$my_age = 2;
print("\$my_age: " . $my_age . "\n");
print("((\$my_age <= 3) || (\$my_age >= 65)): " . "\n");
print("You should stay home" . "\n") if (($my_age <= 3) || ($my_age >= 65));

$my_age = 89;
print("\$my_age: " . $my_age . "\n");
print("((\$my_age <= 3) || (\$my_age >= 65)): " . "\n");
print("You should stay home" . "\n") if (($my_age <= 3) || ($my_age >= 65));

$my_age = 2;
print("\$my_age: " . $my_age . "\n");
print("((\$my_age <= 3) or (\$my_age >= 65)): " . "\n");
print("You should stay home" . "\n") if (($my_age <= 3) or ($my_age >= 65));

$my_age = 89;
print("\$my_age: " . $my_age . "\n");
print("((\$my_age <= 3) or (\$my_age >= 65)): " . "\n");
print("You should stay home" . "\n") if (($my_age <= 3) or ($my_age >= 65));

print("\n# Logical NOT [!, not]\n");

$can_drive = 0;
print("\$can_drive:" . print_bool($can_drive) . "\n");
print("!\$can_drive:" . print_bool(!$can_drive) . "\n");

$can_drive = 1;
print("\$can_drive:" . print_bool($can_drive) . "\n");
print("!\$can_drive:" . print_bool(!$can_drive) . "\n");

$can_drive = 0;
print("\$can_drive:" . print_bool($can_drive) . "\n");
print("not \$can_drive:" . print_bool(not $can_drive) . "\n");

$can_drive = 1;
print("\$can_drive:" . print_bool($can_drive) . "\n");
print("not \$can_drive:" . print_bool(not $can_drive) . "\n");

File: codes/r/conditionals.r
# Relational Operators / Comparison Operators:
# equal to (==)
# not equal to (!=)
# greater than (>)
# less than (<)
# greater then or equal to (>=)
# less then or equal to (<=)

# Logical Operators:
# Logical AND (&&)
# Logical OR (||)
# Logical NOT (!)

CORRECT_ANSWER <- 100

cat("\n# basic conditional control flow\n")

myAge <- 10
cat(paste(sep = "", "myAge: ", myAge, "\n"))
if (myAge > 24) {
    cat("you are old\n")
} else if (myAge > 17) {
    cat("you are young\n")
} else {
    cat("you are under age\n")
}

cat("\n# equal to (==)\n")

myAnswer <- "100"
cat(paste(sep = "", "myAnswer: ", myAnswer, "\n"))
cat(paste(sep = "", "CORRECT_ANSWER: ", CORRECT_ANSWER, "\n"))
cat("(myAnswer == CORRECT_ANSWER):\n")
if (myAnswer == CORRECT_ANSWER) {
    cat("Your answer is correct\n")
} else {
    cat("Your answer is wrong\n")
}

cat("\n# not equal to (!=)\n")

myAnswer <- "25"
cat(paste(sep = "", "myAnswer: ", myAnswer, "\n"))
cat(paste(sep = "", "CORRECT_ANSWER: ", CORRECT_ANSWER, "\n"))
cat("(myAnswer != CORRECT_ANSWER):\n")
if (myAnswer != CORRECT_ANSWER) {
    cat("Your answer is wrong\n")
} else {
    cat("Your answer is correct\n")
}

cat("\n# greater than (>)\n")

myAge <- 70
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("(myAge > 50):\n")
if (myAge > 50) cat("You are old\n")

cat("\n# less than (<)\n")

myAge <- 16
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("(myAge < 20):\n")
if (myAge < 20) cat("You are young\n")

cat("\n# greater then or equal to (>=)\n")

myAge <- 73
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("(myAge >= 65):\n")
if (myAge >= 65) cat("You are allowed to retire\n")

cat("\n# less then or equal to (<=)\n")

myAge <- 14
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("(myAge <= 16):\n")
if (myAge <= 16) cat("You are not allowed to drive\n")

cat("\n# Logical AND (&&)\n")

myAge <- 17
cat(paste(sep = "", "myAge: ", myAge, "\n"))
hasDrivingLicense <- FALSE
cat(paste(sep = "", "hasDrivingLicense: ", hasDrivingLicense, "\n"))
cat("((myAge >= 17) && (hasDrivingLicense == TRUE)):\n")
if ((myAge >= 17) && (hasDrivingLicense == TRUE)) {
    cat("You are allowed to drive\n")
} else {
    cat("You are not allowed to drive\n")
}

myAge <- 17
cat(paste(sep = "", "myAge: ", myAge, "\n"))
hasDrivingLicense <- TRUE
cat(paste(sep = "", "hasDrivingLicense: ", hasDrivingLicense, "\n"))
cat("((myAge >= 17) && (hasDrivingLicense == TRUE)):\n")
if ((myAge >= 17) && (hasDrivingLicense == TRUE)) {
    cat("You are allowed to drive\n")
} else {
    cat("You are not allowed to drive\n")
}

cat("\n# Logical OR (||)\n")

myAge <- 2
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("((myAge <= 3) || (myAge >= 65)):\n")
if ((myAge <= 3) || (myAge >= 65)) cat("You should stay home\n")

myAge <- 89
cat(paste(sep = "", "myAge: ", myAge, "\n"))
cat("((myAge <= 3) || (myAge >= 65)):\n")
if ((myAge <= 3) || (myAge >= 65)) cat("You should stay home\n")

cat("\n# Logical NOT (!)\n")

canDrive <- FALSE
cat(paste(sep = "", "canDrive: ", canDrive, "\n"))
cat(paste(sep = "", "!canDrive: ", !canDrive, "\n"))

canDrive <- TRUE
cat(paste(sep = "", "canDrive: ", canDrive, "\n"))
cat(paste(sep = "", "!canDrive: ", !canDrive, "\n"))
